tool/lib/commands/shared.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:args/command_runner.dart';
+
+extension CommandExtension on Command {
+  void logStatus(String log) {
+    print('[$name] $log');
+  }
+}
+
+extension BuildCommandArgsExtension on ArgParser {
+  void addBulidModeOption() {
+    addOption(
+      BuildCommandArgs.buildMode.name,
+      allowed: ['debug', 'profile', 'release'],
+      defaultsTo: 'release',
+      help: 'The build mode to use for the DevTools web app. This should only'
+          ' be "debug" or "profile" for the purpose of local development.',
+    );
+  }
+
+  void addPubGetFlag() {
+    addFlag(
+      BuildCommandArgs.pubGet.flagName,
+      negatable: true,
+      defaultsTo: true,
+      help: 'Whether to run `devtools_tool pub-get --only-main` before building'
+          ' the DevTools web app.',
+    );
+  }
+
+  void addUpdatePerfettoFlag() {
+    addFlag(
+      BuildCommandArgs.updatePerfetto.flagName,
+      negatable: false,
+      defaultsTo: false,
+      help: 'Whether to update the Perfetto assets before building DevTools.',
+    );
+  }
+
+  void addUseLocalFlutterFlag() {
+    addFlag(
+      BuildCommandArgs.useLocalFlutter.flagName,
+      negatable: false,
+      defaultsTo: false,
+      help: 'Whether to use the Flutter SDK on PATH instead of the Flutter SDK '
+          'contained in the "tool/flutter-sdk" directory.',
```

Comment:
I didn't realise we had this flag elsewhere, but the name seems confusing to me. The word "local" would make me think it uses the version local to DevTools (`tools/flutter-sdk`) though this text suggests it's the other way around. Is it worth making more explicit (`--use-flutter-from-path` or `--use-flutter-from-devtools` or something) with an alias for the old name?

(it's possible we can remove the change I added to support FLUTTER_ROOT recently if we have this - I added it specifically to force using the version from `tool/flutter-sdk` because I was getting the one from `PATH` that I didn't want).

Reply:
changed to --use-flutter-from-path to make this more clear.

Reply:
Cool, thanks! I'll make a note to look back at the FLUTTER_ROOT change I made.. I should be able to remove it if we're generally defaulting to tool/flutter-sdk by default. If there are places that's not happening, I'll fix them up.